In [43]:
!pip install sentence-transformers
from sentence_transformers import SentenceTransformer

In [44]:
model = SentenceTransformer("all-MiniLM-L6-v2")

Define the first set of sentences and get its embeddings using **model.encode()** method.

In [45]:
sentences1 = ["I am a boy", "I like music", "This movie was awesome"]

In [46]:
embeddings1 = model.encode(sentences1, convert_to_tensor=True)

In [47]:
embeddings1

tensor([[-0.0648,  0.1115, -0.0009,  ...,  0.0043, -0.0082, -0.1335],
        [-0.0072, -0.0693,  0.0058,  ...,  0.0597,  0.0734, -0.0376],
        [-0.1320,  0.0089, -0.0492,  ...,  0.0444,  0.0057, -0.0286]])

The define the second set of sentences and get its embeddings.

In [48]:
sentences2 = ["He is a boy", "He likes music", "That dog is so ugly"]

In [49]:
embeddings2 = model.encode(sentences2, convert_to_tensor=True)

In [50]:
embeddings2

tensor([[-0.0743,  0.1551, -0.0569,  ..., -0.0122,  0.0212, -0.0797],
        [ 0.0725,  0.0083,  0.0374,  ...,  0.0592,  0.0626, -0.0019],
        [-0.0156,  0.0659,  0.0607,  ..., -0.0024,  0.0495,  0.0512]])

To compare both the embeddings use the cosine formula, as it calculates the angle between any two vector in space.

In [51]:
from sentence_transformers import util

In [52]:
cosine_scores = util.cos_sim(embeddings1, embeddings2)

In [53]:
cosine_scores

tensor([[0.7027, 0.1997, 0.1049],
        [0.1372, 0.6581, 0.0910],
        [0.1497, 0.1047, 0.1928]])

In [54]:
for i in range(len(sentences1)):
    print("{} \t\t {} \t\t Score: {:.4f}".format(sentences1[i],
                                                 sentences2[i],
                                                 cosine_scores[i][i]))

I am a boy 		 He is a boy 		 Score: 0.7027
I like music 		 He likes music 		 Score: 0.6581
This movie was awesome 		 That dog is so ugly 		 Score: 0.1928


The first two sentences in both the set are very similar hence high scores (0.70727 and 0.6581), the last sentences on both the sets are not at all similar hence low score (0.1928)